src/test/java/org/terracotta/statistics/simulation/latency/LatencyHistogramSimulator.java

```diff
@@ -0,0 +1,287 @@
+  private static final Duration COLLECTOR_INTERVAL = ofSeconds(10);
+  private static final Duration MEAN_LATENCY = ofMillis(100);
+
+  // Sorted list of samples, by time
+  private static List<LongSample> OPERATIONS;
+
+  @Parameters(name = "{index}: win={0}_buck={1}_phi={2}_frame={3}_pts={4}")
+  public static Iterable<Object[]> data() {
+
+    // SETTINGS FOR: histograms
+    double[] PHIS = {0.3, 0.63, 0.7, 0.8, 2.0}; // 0.7 == BarSplittingBiasedHistogram.DEFAULT_PHI
+    int[] BUCKETS = {10, 20, 50};
+    Duration[] WINDOWS = {ofMinutes(1), ofMinutes(5), ofMinutes(10), ofMinutes(30), ofMinutes(60), ofMinutes(120)};
+
+    // SETTINGS FOR: graph summary in management console
+    int[] POINTS = {20, 30, 40}; // number of points to display for each graph
+    Duration[] TIME_FRAMES = {ofMinutes(5), ofMinutes(10), ofMinutes(30), ofMinutes(60)}; // time frame of each graph
+
+    // Generates Junit's parameters
+    return Stream.of(TIME_FRAMES).flatMap(tf ->
+        IntStream.of(POINTS).boxed().flatMap(points ->
+            DoubleStream.of(PHIS).boxed().flatMap(phi ->
+                IntStream.of(BUCKETS).boxed().flatMap(buckets ->
+                    Stream.of(WINDOWS).map(window ->
+                        new Object[]{window, buckets, phi, tf, points}
+                    )))))
+        //.filter(oo -> oo[4].equals(40)) // TO FILTER OUT SOME CASES
+        //.limit(40)
+        .collect(Collectors.toList());
+  }
+
+  @BeforeClass
+  public static void prepareFolder() {
+    if (OUTPUT.exists()) {
+      if (OUTPUT.isFile()) {
+        OUTPUT.delete();
+      } else {
+        Stream.of(OUTPUT.listFiles()).parallel().forEach(File::delete);
+      }
+    }
+    OUTPUT.mkdirs();
+    System.out.println("OUTPUT: " + OUTPUT.getAbsolutePath() + lineSeparator());
+  }
+
+  @BeforeClass
+  public static void simulateOperations() {
+    long seed = System.nanoTime();
+    System.out.println("SEED: " + seed); // keep this in case test fails so that we can reproduce the failure
+
+    OPERATIONS = generateLatencies(new Random(seed), 15, ofMinutes(90), MEAN_LATENCY.toNanos());
+    System.out.println("SIMULATED LATENCIES: " + OPERATIONS.size() + lineSeparator());
+
+    Map<String, Percentile> percentiles = getPercentiles(OPERATIONS, nsToMicros());
+    System.out.println("LATENCIES PERCENTILES (μs): " + percentiles + lineSeparator());
+
+    // write data, graph and stats on disk
+    writeCSV(new File(OUTPUT, "latencies-data.csv"), OPERATIONS);
+    writeGraph(
+        new File(OUTPUT, "latencies-graph.png"),
+        OPERATIONS,
+        percentiles,
+        DomainOrder.ASCENDING,
+        "time (s)", "latency (μs)", FHD,
+        nsToSecs(),
+        nsToMicros());
+  }
+
+  @AfterClass
+  public static void openReport() throws IOException {
+    new ProcessBuilder("open", "src/test/resources/Latencies.html").start();
```

Comment:
MacOS specific?  Isn't this path wrong now too?

Reply:
yes path is wrong now ;-) I'll fix that.
For the MacOS thing, this is why I do not wait for the process because I know it will fail on some machines. It will work for most of us but if somebody is running windows and __really__ want to have this think working, then he'll fix it!